content/outcomes/application-development-rhythm/dev-flow.md

```diff
@@ -0,0 +1,227 @@
+The *Backlog* is a curated list from which the developer will pull their
+work.
+
+The work is specified as discrete *Stories* that are:
+
+- Specifications of *what* to build (or fix).
+- Estimated by the team.
+- Independent of each other.
+- Prioritized.
+- Sized small enough to be integrated daily.
+
+Note that product teams will typically organize work around features or
+fixes,
+either of which may comprise work that is not small enough to fit in the
+confines of a single story.
+
+Agile frameworks specify course grained work as an *Epic* or *Theme*,
+but neither the Epic or Theme directly specify the work executed by the
+developer.
+Stories may be tied to Epics (or Themes) to keep context of the work to
+a specific feature or fix.
+
+Outside of the scope of the developer workflow,
+use of Epics and their relationships to stories are important when
+specifications are not well known,
+or when refining in-process work.
+
+The input for an Agile developer workflow is a single story.
+
+## Software Delivery
+
+The output of the Agile developer's workflow is functioning software
+that is pulled into a downstream
+*[Continuous Delivery](https://tanzu.vmware.com/developer/guides/ci-cd/ci-cd-what-is/#what-is-cd)*
+*Pipeline* for purposes of being able to deliver features or fixes to
+production at any time.
+
+The *Continuous Delivery Pipeline* may involve various manual and/or
+automated steps for testing,
+verification,
+oversight compliance,
+staging,
+or perhaps a
+[secure pipeline towards production](https://www.thoughtworks.com/insights/articles/towards-a-secure-path-to-production).
+
+*Note*:
+There is much more to this topic that is outside the scope of this
+article.
+
+## The Agile Developer Workflow
+
+While the story provides some form of specification or requirement,
+it is up to the developer to translate the specification of the story
+into one or more tests and their associated solutions.
+
+The Agile development workflow compresses design,
+construction,
+and a substantial amount of testing into a story's implementation work.
+
+This places more burden on the developer and the need for discipline in
+the developer workflow.
+
+## Flow Summary
+
+An Agile developer's daily flow looks like this:
+
+1.  Take the highest priority story from the backlog.
+1.  Review the story details and its desired outcome.
+    The story may be for building new features,
+    fixing defects,
+    or investigation if the problem/solution domain is not well known.
+    If the story is investigative,
+    the developer may use a
+    *[Spike](https://www.leadingagile.com/2016/09/whats-a-spike-who-should-enter-it-how-to-word-it/)*
+    as a way to timebox effort to better understand a problem and/or its
+    solution.
+1.  Iterate on a series of steps using one or more *feedback loops* to
+    achieve the outcome of the story.
+    Developers will spend much of their time writing code,
+    using methods such as *Test Driven Development*,
```

Comment:
Consider linking to TDD article in the Fundamentals course.

Reply:
@joemoore thanks for feedback, fixed